src/Catalyst.Protocol/Transaction/PublicEntry.cs

```diff
@@ -33,13 +34,41 @@ public partial class PublicEntry
 
         public bool IsValid()
         {
-            if (Amount == null || Amount.IsEmpty || Amount.All(b => b == default))
+            // make it a constant - MIN_GAS_LIMIT
+            if (GasLimit < 21000)
+            {
+                return false;
+            }
+
+            var isTimestampValid = Timestamp != default(Timestamp) && Timestamp != new Timestamp();
+            if (!isTimestampValid)
             {
-                Logger.Debug("{field} cannot be 0", nameof(Amount));
+                Logger.Debug("{timestamp} cannot be null or 0.");
                 return false;
             }
 
             return true;
+
+            // TODO: reconsider signature
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4872747)